src/index.ts

```diff
@@ -76,6 +82,41 @@ export default class Api {
     this.timeTrackings = new Timetrackings(this.config);
     this.users = new Users(this.config);
     this.userStatusOverviews = new UserStatusOverviews(this.config);
+
+    axios.interceptors.response.use(
+      (res) => {
+        return res;
+      },
+      async (error) => {
+        if (this.config.autoRefreshToken) {
+          const untouchedRequest = error.config;
+
+          if (error.response.status === 401 && untouchedRequest.url.indexOf('auth/oauth2/token') !== -1) {
+            return Promise.reject(error);
+          }
+          if (error.response.status === 401 && !untouchedRequest._retry) {
+            untouchedRequest._retry = true;
+
+            if (!this.state.refreshingToken) {
```

Comment:
To avoid unnecessary token refreshes I believe we need to check whether the untouched Request contains the current tokens. If not, no need to refresh tokens, just use the current tokens from `this.authentication.tokens` to retry the request

Reply:
As discussed this approach bring us additional work.
Will leave it as it is, and have in consideration as improvement later on.